src/python/pants/init/specs_calculator.py

```diff
@@ -98,10 +101,12 @@ def create(
                 sources=tuple(changed_options.changed_files(scm=scm)),
                 dependees=changed_options.dependees,
             )
-            (changed_addresses,) = session.product_request(ChangedAddresses, [changed_request])
-            logger.debug("changed addresses: %s", changed_addresses.addresses)
+            (changed_addresses,) = session.product_request(
+                ChangedAddresses, [Params(changed_request, options_bootstrapper)]
```

Comment:
FYI we must pass `options_bootstrapper` because dep inference requires it. We now use dep inference when considering `--changed-dependees`.